lib/lambda_ethereum_consensus/p2p/block_downloader.ex

```diff
@@ -135,18 +157,38 @@ defmodule LambdaEthereumConsensus.P2P.BlockDownloader do
     end
   end
 
-  defp decode_response(response) do
-    {_size, rest} = P2P.Utils.decode_varint(response)
+  @spec decode_chunks([binary()]) :: {:ok, [SszTypes.SignedBeaconBlock.t()]} | {:error, binary()}
+  defp decode_chunks(chunks) do
+    results =
+      chunks
+      |> Enum.map(&decode_chunk/1)
+
+    if Enum.all?(results, fn
+         {:ok, _} -> true
+         _ -> false
+       end) do
```

Comment:
We can leave it for a later PR, but we should check each block independently. The peer could have only _some_ of the blocks we requested and return an error on those missing.

Reply:
Hmm yeah, this is tricky. The problem is that then the caller would have to process the response the data and have potentially complex logic to deal with the case in which some blocks are there and some are not

Reply:
I opened #364 for this